lib/secretsscanner/scaner/scan_test.go

```diff
@@ -190,3 +194,24 @@ func writeEncryptedKeyWithoutPubFile(t *testing.T, dir string) []*accessgraphsec
 
 	return []*accessgraphsecretsv1pb.PrivateKey{key}
 }
+
+func writeInvalidKeys(t *testing.T, dir string) []*accessgraphsecretsv1pb.PrivateKey {
+	t.Helper()
+
+	// Write invalid keys to the directory.
+	err := os.WriteFile(filepath.Join(dir, "file-with-short-size"), []byte("invalid-key"), 0o644)
+	require.NoError(t, err)
+
+	err = os.WriteFile(filepath.Join(dir, "invalid-key-valid-headers"), []byte(
+		`-----BEGIN OPENSSH PRIVATE KEY-----\n
+trash\n
+-----END OPENSSH PRIVATE KEY-----\n`), 0o644)
+	require.NoError(t, err)
+
+	err = os.WriteFile(filepath.Join(dir, "invalid-key-invalid-header"), []byte(
+		`abcefg-----BEGIN OPENSSH PRIVATE KEY-----\n
```

Comment:
Since we have confirmed that this errors, should we change our bytes.Contains check to bytes.HasPrefix? (Scanner.readFileIfSSHPrivateKey)

Reply:
I wanted to avoid that because pem decode supports `\n-----BEGIN OPENSSH PRIVATE KEY-----`

Reply:
`bytes.HasPrefix(bytes.TrimLeft(b, "\n "), "-----BEGIN OPENSSH PRIVATE KEY----")`

or, if you'd rather not that, leave a comment explaining why a Contains and not a HasPrefix.

I would also add the `\n-----BEGIN OPENSSH PRIVATE KEY-----` to the scenarios.